### PR TITLE
Several UI fixes

### PR DESCRIPTION
**Pull request by DRSDavidSoft**
According to the individual commits, minor fixes made to several parts of the UI.

### CODE REVIEW

**Comment by DRSDavidSoft** [comment]
@RekkyRek Hi, why the tests are failing?

**Comment by hex2f** [comment]
@DRSDavidSoft are you using eslint to check linting before push?

**Comment by tahnik** [comment]
@DRSDavidSoft 
Please check the log of Appvoyer or Travis.

Here is the error: 
C:\projects\devrantron\app\src\main\js\components\auth\login.js
  47:14  error  Property should be placed on a new line  react/jsx-first-prop-new-line

By the way, welcome to the project and thank you!

File: app/src/main/js/components/auth/login.js [hunk 1]
```diff
@@ -44,7 +44,12 @@ class Login extends Component {
         <div className="auth_image" >
           <img alt="devrant" src="./res/images/devrant_sidebar.png" />
         </div>
-        <div className="auth_form" >
+        <div className="auth_form"
```
Comment (tahnik) [inline app/src/main/js/components/auth/login.js]: We really need this, but this doesn't work for some reason. Can you check?

Edit: The reason is this is a div, not a form. So onSubmit won't be triggered.

File: app/src/main/res/css/modules/bottom_bar.sass [hunk 2]
```diff
@@ -2,7 +2,7 @@
   font-size: 1.5rem
   display: flex
   background-color: #40415A
-  padding: 0.5rem
+  padding: 0.8rem 1rem
```
Comment (tahnik) [inline app/src/main/res/css/modules/bottom_bar.sass]: @Dacexi can you review this one?

File: app/src/main/res/css/modules/column.sass [hunk 3]
```diff
@@ -39,13 +39,15 @@
...
-    overflow-x: hidden
+    position: relative
+    overflow: auto
+    overflow-x: visible
     transform: translateZ(0)
     -webkit-transform: translateZ(0)
-    padding: 0rem 0rem 1rem 0rem
-    height: calc(100vh - 2.8rem - 1.5rem)
+    padding: 0 10px
+    height: 100vh
```
Comment (tahnik) [inline app/src/main/res/css/modules/column.sass]: Don't make it 100vh please. Keep some padding in the bottom. This can cause side effects.

File: app/src/main/res/css/modules/column.sass [hunk 4]
```diff
@@ -39,13 +39,15 @@
     margin-top: 5rem
     font-size: 1.3rem
   .items_container
-    overflow-x: hidden
-    overflow-y: scroll
-    overflow-x: hidden
+    position: relative
+    overflow: auto
```
Comment (tahnik) [inline app/src/main/res/css/modules/column.sass]: Sometimes some rants cause overflow-x issues. It would be better to keep it the way it is.

File: app/src/main/res/css/modules/column.sass [hunk 5]
```diff
@@ -39,13 +39,15 @@
...
+    overflow-x: visible
     transform: translateZ(0)
     -webkit-transform: translateZ(0)
-    padding: 0rem 0rem 1rem 0rem
-    height: calc(100vh - 2.8rem - 1.5rem)
+    padding: 0 10px
```
Comment (tahnik) [inline app/src/main/res/css/modules/column.sass]: This is causing the column not to align with the topbar. Is this what you intended? I don't think that looks nice. Maybe @Dacexi can confirm it.

File: app/src/main/res/css/modules/compact_user_card.sass [hunk 6]
```diff
@@ -51,8 +52,8 @@
       cursor: pointer
   .logout
     position: absolute
-    bottom: 5px
-    left: 10px
+    bottom: 10px
```
Comment (tahnik) [inline app/src/main/res/css/modules/compact_user_card.sass]: This is causing the shadow in the back to misalign

File: app/src/main/res/css/modules/item_card.sass [hunk 7]
```diff
@@ -41,7 +40,7 @@
     width: 100%
 .item_card.shadow
   &:hover
-    box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23)
+    box-shadow: 0 2px 0px rgba(0, 0, 0, 0.19), 0 3px 4px rgba(0, 0, 0, 0.23)
```
Comment (tahnik) [inline app/src/main/res/css/modules/item_card.sass]: Don't think it's looking nice either. Looks more like a black border than a shadow.

**Comment by DRSDavidSoft** [comment]
@tahnik Great! Why using two CI services may I ask?

**Comment by tahnik** [comment]
@DRSDavidSoft Travis doesn't support Windows. We use AppVeyor for Windows and Travis for macOS, Linux

**Comment by DRSDavidSoft** [comment]
@tahnik Can you review the PR?

**Comment by tahnik** [comment]
@DRSDavidSoft I did, check above

**Comment by tahnik** [comment]
@DRSDavidSoft have you found the reviews? Just making sure!

File: app/src/main/res/css/modules/compact_user_card.sass (hunk 6 above)
Comment (hex2f) [inline app/src/main/res/css/modules/compact_user_card.sass]: change to 0.25rem

File: app/src/main/res/css/modules/compact_user_card.sass [hunk 8]
```diff
@@ -51,8 +52,8 @@
...
+    left: 1.5rem
```
Comment (hex2f) [inline app/src/main/res/css/modules/compact_user_card.sass]: change to 0.5rem instead. Looks a bit odd when it's so high up.

File: app/src/main/res/css/modules/compact_user_card.sass (hunk 6 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/compact_user_card.sass]: Done. Initially I used `px` instead of `rem` because the previous commit also used it.
I suppose most of the units should be using `rem` here, right?

File: app/src/main/res/css/modules/compact_user_card.sass (hunk 8 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/compact_user_card.sass]: Are you referring to the `left` property? I set it to `0.5rem`, but could you comment on the suitable values for both `left` and `bottom` properties?

**Comment by tahnik** [comment]
@DRSDavidSoft could you give me your email so I could add you in slack? It's easier to talk over there. Thanks

**Comment by DRSDavidSoft** [comment]
@tahnik Thanks, it's David@Refoua.me

**Comment by tahnik** [comment]
@DRSDavidSoft I sent you the request. Please join.

**Comment by DRSDavidSoft** [comment]
Thank you, @tahnik. I joined. What about this PR?

File: app/src/main/js/components/auth/login.js [hunk 9]
```diff
@@ -44,7 +44,13 @@ class Login extends Component {
         <div className="auth_image" >
           <img alt="devrant" src="./res/images/devrant_sidebar.png" />
         </div>
-        <div className="auth_form" >
+        <div
```
Comment (tahnik) [inline app/src/main/js/components/auth/login.js]: Please remove this as this is not working :)

**Comment by DRSDavidSoft** [comment]
The PR is conflicting with the edits. Could you review & merge it before more conflicts to show up?
Also what does the `keydown` event handler do?

**Comment by tahnik** [comment]
I resolved the conflicts. Can you please look at the reviews here: https://github.com/tahnik/devRantron/pull/136/files? I won't be able to merge this before you fix these stuff.

**Comment by DRSDavidSoft** [comment]
@tahnik If you've reviewed/commented on my commits, I don't seem to find anything from you.
http://imgur.com/a/11CrO

**Comment by tahnik** [comment]
yeah, because I just started to push to your branch and fix those reviews @DRSDavidSoft

**Review by tahnik — CHANGES_REQUESTED** [review]

**Comment by tahnik** [comment]
fuck me sorry. I forgot to request the changes

**Comment by tahnik** [comment]
can you see the reviews now @DRSDavidSoft

File: app/src/main/res/css/modules/user_card.sass [hunk 10]
```diff
@@ -1,16 +1,17 @@
 .user_card
   position: absolute
-  width: 25rem
+  width: calc(100% - 2rem)
   background-size: cover !important
   background-repeat: no-repeat !important
-  padding: 1rem 1rem 0rem 1rem
+  box-shadow: 0 2px 0px rgba(0, 0, 0, 0.19), 0 3px 4px rgba(0, 0, 0, 0.23)
+  padding: 1rem 1rem 2rem 1rem
```
Comment (tahnik) [inline app/src/main/res/css/modules/user_card.sass]: This is totally messed up. http://imgur.com/a/j0S0L

File: app/src/main/js/components/auth/login.js (hunk 1 above)
Comment (DRSDavidSoft) [reply app/src/main/js/components/auth/login.js]: Why not change it to a `form`? It makes sense
Otherwise, we should write a handler on both fields to check `e.which == 13` on `keypress`, which I don't recommend

File: app/src/main/js/components/auth/login.js (hunk 9 above)
Comment (DRSDavidSoft) [reply app/src/main/js/components/auth/login.js]: Can `.auth_form` be an actual `<form>`? :)

File: app/src/main/res/css/modules/bottom_bar.sass (hunk 2 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/bottom_bar.sass]: @RekkyRek I'd be grateful if you did!

File: app/src/main/js/components/auth/login.js (hunk 1 above)
Comment (tahnik) [reply app/src/main/js/components/auth/login.js]: We have recently fixed it using keyDownEvent. so don't worry about this one.

File: app/src/main/res/css/modules/column.sass (hunk 4 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/column.sass]: Nope, it's not. I think I know what you're referring to. Instead of `overflow-x`, this should be done:
https://stackoverflow.com/q/3058866/1454514

or better - https://stackoverflow.com/a/18706058/1454514

TL;DR: `word-wrap: break-word;` & `word-break: break-all;`

Comment (tahnik) [reply app/src/main/res/css/modules/column.sass]: You can add word-break: break-all; but it is still better to keep overflow-x hidden.

File: app/src/main/res/css/modules/column.sass (hunk 5 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/column.sass]: @tahnik Can you post a screenshot from before and after please?

Comment (tahnik) [reply app/src/main/res/css/modules/column.sass]: I fixed it in another commit.

**Review by tahnik — APPROVED** [review]

**Comment by tahnik** [comment]
@DRSDavidSoft does everything looks good to you now?

File: app/src/main/res/css/modules/column.sass (hunk 4 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/column.sass]: @tahnik I hear you, but it causes problems with `box-shadow`. If it's going to be `hidden`, there should be enough `padding` for the `box-shadow` IMHO.

File: app/src/main/res/css/modules/column.sass (hunk 3 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/column.sass]: @tahnik Tested with some other situations, I agree.

File: app/src/main/res/css/modules/item_card.sass (hunk 7 above)
Comment (DRSDavidSoft) [reply app/src/main/res/css/modules/item_card.sass]: @tahnik, OK, let's keep the previous one, but please make sure it has enough `padding`.

File: app/src/main/res/css/modules/column.sass (hunk 4 above)
Comment (tahnik) [reply app/src/main/res/css/modules/column.sass]: @DRSDavidSoft break-all causes some other problems. For example, it also breaks words which look horrible. box-shadow looks fine now so have a look.

**Comment by DRSDavidSoft** [comment]
@tahnik Seems good! I was wondering why I didn't get any reviews from you. Let's merge it :)
